src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -1603,6 +1603,97 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
   </Target>
 
+  <!--
+    ====================================================================================
+                                        _GetProjectReferencePlatformProperties
+
+    If a project is opted in via $(EnableDynamicPlatformResolution), this target builds the GetPlatforms
+    target of all existing references to get a list of all possible platforms of the referenced projects. 
+    Calls the GetCompatiblePlatform task to determine the most compatible platform for each project.
+    This prevents overbuilding a project when 'AnyCPU' is available, for example.
+
+    ======================================================================================
+  -->
+
+  <Target Name="_GetProjectReferencePlatformProperties"
+          BeforeTargets="_GetProjectReferenceTargetFrameworkProperties"
+          Condition="'$(EnableDynamicPlatformResolution)' == 'true'
+                     and '$(BuildingInsideVisualStudio)' != 'true'
+                     and '@(_MSBuildProjectReferenceExistent->Count())' != '0'">
+
+    <!-- Allow preset SetPlatform to override this operation -->
+    <ItemGroup>
+      <_MSBuildProjectReferenceExistent Condition="'%(_MSBuildProjectReferenceExistent.SetPlatform)' != ''">
+        <SkipGetPlatformProperties>true</SkipGetPlatformProperties>
+      </_MSBuildProjectReferenceExistent>
+    </ItemGroup>
+
+    <MSBuild Projects="@(_MSBuildProjectReferenceExistent)"
+        Targets="GetPlatforms"
+        BuildInParallel="$(BuildInParallel)"
+        ContinueOnError="!$(BuildingProject)"
+        RemoveProperties="%(_MSBuildProjectReferenceExistent.GlobalPropertiesToRemove);TargetFramework;RuntimeIdentifier$(_GlobalPropertiesToRemoveFromProjectReferences)"
+        Condition="'%(_MSBuildProjectReferenceExistent.SkipGetPlatformProperties)' != 'true'"
+        SkipNonexistentTargets="true">
+      <Output TaskParameter="TargetOutputs" ItemName="_ProjectReferencePlatformPossibilities" />
+    </MSBuild>
+
+    <!-- Managed Platform "source of truth" is $(PlatformTarget). For cpp it's $(Platform) -->
+    <PropertyGroup>
+      <ParentPlatform>$(PlatformTarget)</ParentPlatform>
+      <ParentPlatform Condition="'$(MSBuildProjectExtension)' == '.vcxproj' or '$(MSBuildProjectExtension)' == '.nativeproj'">$(Platform)</ParentPlatform>
+    </PropertyGroup>
+
+    <!-- If we're looking at a c++ project from a managed project, map managed platforms to native platforms. -->
+    <ItemGroup>
+      <_ProjectReferencePlatformPossibilities Condition="'$(MSBuildProjectExtension)' != '.vcxproj' and '$(MSBuildProjectExtension)' != '.nativeproj' and '%(_ProjectReferencePlatformPossibilities.IsVcxOrNativeProj)' == 'true'">
+        <!-- Taken from https://docs.microsoft.com/visualstudio/msbuild/assignprojectconfiguration-task -->
+        <PlatformLookupTable>AnyCPU=Win32;x86=Win32;x64=x64</PlatformLookupTable>
```

Comment:
cc @yuehuang010 is anycpu->win32 acceptable here as a default?

Also note to self: this property should not assign a value to `PlatformLookupTable` if it's already set (to allow user customization)

Reply:
How would vcxproj --> csproj work?  What platform will vcx set for cs project?  AnyCPU or x86?
Likewise, csproj --> vcxproj --> csproj, will have platform AnyCPU --> Win32 --> AnyCPU or x86?

Reply:
> How would vcxproj --> csproj work? What platform will vcx set for cs project? AnyCPU or x86?

Would love your input on that actually. If you have any vcx->cs reference you need to define `PlatformLookupTable` in that `vcxproj` or somewhere for that project. Is there any sort of acceptable "default" platform lookup for vcx->cs?

> Likewise, csproj --> vcxproj --> csproj, will have platform AnyCPU --> Win32 --> AnyCPU or x86?

In this example of A->B->C, there's currently no way for C to know that A built with AnyCPU. The path forward here is to define the `PlatformLookupTable` in B so that it knows what it should point a Win32 build to.

ALso note to self: x64=x64 is not necessary. The logic already exists in the class.

Reply:
IMO AnyCPU->Win32 should never happen automatically. C# doesn't even allow a C#->C# p2p from AnyCPU->x86 (without a warning, at least). AnyCPU is supposed to mean you are architecture agnostic, but you aren't if you reference an architecture-specific project.
vcxproj->csproj should map to arch-specific if there is a match (e.g. win32->x86, x64->x64) and fallback to AnyCPU otherwise (e.g. win32->anycpu).
> In this example of A->B->C, there's currently no way for C to know that A built with AnyCPU.
It shouldn't know. We don't want C influenced by A. When I build B with a given platform it should build the same (including with C) whether or not B was the top-level build or built as a P2P target of A. Given B is Win32, C should be build with x86 if there is one, or AnyCPU otherwise.